rclpy/test/test_node.py

```diff
@@ -838,6 +838,162 @@ def test_node_set_parameters_rejection(self):
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple1[0][0],
+                    value=reject_list_parameter_tuple1[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[1][0],
+                    value=reject_list_parameter_tuple1[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[2][0],
+                    value=reject_list_parameter_tuple1[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+
+        # Adding more parameters.
+        success_parameters_list_tuple = [
+            (
+                's1',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                's2',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                's3',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+        self.node.declare_parameters('', success_parameters_list_tuple)
+        result = self.node._set_parameters(
+            [
+                Parameter(
+                    name=success_parameters_list_tuple[0][0],
+                    value=success_parameters_list_tuple[0][1]
+                ),
+                Parameter(
+                    name=success_parameters_list_tuple[1][0],
+                    value=success_parameters_list_tuple[1][1]
+                ),
+                Parameter(
+                    name=success_parameters_list_tuple[2][0],
+                    value=success_parameters_list_tuple[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertTrue(result[0].successful)
+
+    def test_node_add_on_set_paramter_callback(self):
+        # Add callbacks to the list of callbacks.
+
+        callbacks = [
+            self.reject_parameter_callback,
+            self.reject_parameter_callback_1,
+            self.reject_parameter_callback_2,
+            self.reject_parameter_callback_3
+        ]
+        for callback in callbacks:
+            self.node.add_on_set_parameters_callback(callback)
+
+        for callback in callbacks:
+            self.assertTrue(callback in self.node._parameters_callbacks)
+
+    def test_node_remove_from_set_callback(self):
+        # Remove callbacks from list of callbacks.
+        # List of callback one of which is rejecting the paramter value to be set.
```

Comment:
This doesn't seem to be a complete sentence?

Reply:
I removed this sentence sir :)